src/python/example_tensor.py

```diff
@@ -0,0 +1,325 @@
+    else:
+       return self.singa_tensor.shape(axis)
+
+  def ndim(self):
+    return self.singa_tensor.nDim()
+
+  def transpose(self):
+    return self.singa_tensor.transpose()
+
+  def size(self):
+    return self.singa_tensor.Size()
+
+  def memsize(self):
+    return self.singa_tensor.MemSize()
+
+  def reshape(self, shape):
+    self.tuple_shape = shape
+    self.singa_tensor.Reshape(tuple_to_vector(shape))
+
+  def reset_like(self, t):
+    self.singa_tensor.ResetLike(t.singa_tensor)
+
+  def as_type(self, dtype):
+    self.singa_tensor.AsType(dtype)
+
+  def to_device(self, device):
+    self.singa_tensor.ToDevice(device)
+
+  def to_host(self):
+    self.singa_tensor.ToHost()
+
+
+  def set_value(self, x):
+    self.singa_tensor.SetValue(x)
+
+  def copy_data_from_hostptr(self, ptr, num):
+    self.singa_tensor.CopyDataFromHostPtr(ptr, num)
+
+  def copy_data(self, t):
+    self.singa_tensor.CopyData(t.singa_tensor)
+
+  def clone(self):
+    new_t = Tensor(self.tuple_shape, self.device, self.dtype)
+    new_t.singa_tensor = self.singa_tensor.Clone()
+    return new_t
+
+  def matrix_transpose(self):
+    new_t = Tensor(self.tuple_shape, self.device, self.dtype)
+    new_t.singa_tensor = self.singa_tensor.T()
+    return new_t
+
+  # TODO(chonho-02) assign
+  # same as copy_Tensor(t) ???
+  def assign(self, t):
+    self.singa_tensor = self.singa_tensor.Assign_T(t.singa_tensor)
+
+  def __iadd__(self, x):
+    if type(x) == Tensor:
+      self.singa_tensor += x.singa_tensor
+    else:
+      self.singa_tensor += x
+    return self
+
+  def __isub__(self, x):
+    if type(x) == Tensor:
+      self.singa_tensor -= x.singa_tensor
+    else:
+      self.singa_tensor -= x
+    return self
+
+  def __imul__(self, x):
+    if type(x) == Tensor:
+      self.singa_tensor *= x.singa_tensor
+    else:
+      self.singa_tensor *= x
+    return self
+
+  def __idiv__(self, x):
+    if type(x) == Tensor:
+      self.singa_tensor /= x.singa_tensor
+    else:
+      self.singa_tensor /= x
+    return self
+
+  def __add__(lhs, rhs):
+    return call_singa_func(singa.Add_TT, lhs.singa_tensor, rhs.singa_tensor)
+
+  def __sub__(lhs, rhs):
+    return call_singa_func(singa.Sub_TT, lhs.singa_tensor, rhs.singa_tensor)
+
+  def __mul__(lhs, rhs):
+    return call_singa_func(singa.Mul_TT, lhs.singa_tensor, rhs.singa_tensor)
+
+  def __div__(lhs, rhs):
+    return call_singa_func(singa.Div_TT, lhs.singa_tensor, rhs.singa_tensor)
+
+
+  #(TODO) need a method to convert Tensor to numpy array
+  #(TODO) need a method to convert numpy array to Tensor
+
+  def copy_args(self):
+    self.tuple_shape = self.singa_tensor.shape()
+    self.device = self.singa_tensor.device()
+    self.dtype = self.singa_tensor.data_type()
+
+
+''' python functions for global methods in Tensor.h
+'''
+def product(shape):
+  return singa.Product(tuple_to_vector(shape))
+
+def sizeof(dtype):
+  return singa.SizeOf(dtype)
+
+def call_singa_func(_singa_func, *args):
```

Comment:
can we make it private? i.e., users should not call this function.
or need to add comments for it.